modules/core/src/components/declarative-ui/xviz-metric.js

```diff
@@ -75,6 +77,7 @@ class XVIZMetricComponent extends PureComponent {
   constructor(props) {
     super(props);
     this.state = {
+      missingStreams: [],
```

Comment:
You should populated missingStreams in the constructor as well. You may want to just have _getTimeSeries return the missingStreams as well in an object.